modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -898,6 +898,8 @@ public String getName() {
     public final int getCellCount() { return cellCount.get(); }
     public final void setCellCount(int value) {
         resetSizeEstimates();
+        recalculateAndImproveEstimatedSize(2);
+
         cellCount.set(value);
```

Comment:
Can this be implemented in a way that doesn't violate the property contract? Since this property is public API, `setCellCount(int)` should just call `cellCountProperty().set(int)`. Maybe you can extract the composite operation here into a private method and use that instead of `setCellCount(int)`.

`position` has the same problem.

Reply:
good catch! the "additional" stuff typically is done in the properties' invalidated ..

Reply:
Exactly. This ensures consistent behavior regardless of how a property value is set.

Reply:
I should add that this looks like a preexisting problem, but one that would be good to fix if possible.

Reply:
Thanks for catching. I moved the additional code into the invalidated block, at the appropriate places.
Note that this caused some changes in the flow at runtime, where the number of evaluations of `updateItem` changed. Those changes are of constant order (and not linear with the amount of cells), and I made some changes to the test as they depended on hard numbers.